main.lua

```diff
@@ -231,6 +262,27 @@ function renderFieldOutline()
   dottedLine(0.5 * VIRTUAL_WIDTH, BOARD_BORDER_TOP, 0.5 * VIRTUAL_WIDTH, VIRTUAL_HEIGHT, 5, 3)
 end
 
+function logBallOwnership()
+  love.graphics.setColor(getColor('red'))
+  love.graphics.printf('Ball ownership: ' ..BALL_POSSESION, 10, BORDER_LEFT_RIGHT + 10, 300)
+end
+
+function logBallPosition()
+  love.graphics.setColor(getColor('white'))
+  love.graphics.printf('Ball position X: ' ..ball.x, 10, BORDER_LEFT_RIGHT + 20, 300)
+  love.graphics.printf('Ball position Y: ' ..ball.y, 10, BORDER_LEFT_RIGHT + 30, 300)
+  love.graphics.printf('Ball delta X: ' ..ball.dx, 10, BORDER_LEFT_RIGHT + 40, 300)
+  love.graphics.printf('Ball delta Y: ' ..ball.dy, 10, BORDER_LEFT_RIGHT + 50, 300)
+end
+
+-- CONSOLE log
+function renderConsole()
```

Comment:
You could just run game in console and log stuff there using standard lua print method instead doing it in game with love.graphics. If you for some reason still want to have it this way please create separate `console` class and move your logic there.

Reply:
If I am not mistaken, in this case we cannot use lua print method, because the variables are constantly changing values. That means that console would be constantly spammed with new values